experimental/batchmap/cmd/build/mapdemo.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// mapdemo is a simple example that shows how a verifiable map can be
+// constructed in Beam.
+package main
+
+import (
+	"context"
+	"crypto"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/io/filesystem/local"
+	"github.com/apache/beam/sdks/go/pkg/beam/x/beamx"
+
+	"github.com/golang/protobuf/proto"
+
+	"github.com/google/trillian/experimental/batchmap"
+	"github.com/google/trillian/experimental/batchmap/tilepb"
+	"github.com/google/trillian/merkle/coniks"
+)
+
+const hash = crypto.SHA512_256
+
+var (
+	output       = flag.String("output", "", "Output directory in which the tiles will be written.")
+	valueSalt    = flag.String("value_salt", "v1", "Some string that will be smooshed in with the generated value before hashing. Allows generated values to be deterministic but variable.")
+	startKey     = flag.Int64("start_key", 0, "Keys will be generated starting with this index.")
+	keyCount     = flag.Int64("key_count", 1<<5, "The number of keys that will be placed in the map.")
+	treeID       = flag.Int64("tree_id", 12345, "The ID of the tree. Used as a salt in hashing.")
+	prefixStrata = flag.Int("prefix_strata", 1, "The number of strata of 8-bit strata before the final strata. 3 is optimal for trees up to 2^30. 10 is required to import into Trillian.")
+)
+
+func init() {
+	beam.RegisterType(reflect.TypeOf((*mapEntryFn)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*writeTileFn)(nil)).Elem())
+}
+
+func main() {
+	flag.Parse()
+	beam.Init()
+
+	output := filepath.Clean(*output)
+	if output == "" {
+		log.Fatal("No output provided")
```

Comment:
Could use `glog` for consistency with the verifier below, and then switch to `Exitf` in here?

Reply:
Cool, done.